packages/connect-core/src/protocol-connect/trailer-mux.spec.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { trailerMux, trailerDemux } from "./trailer-mux.js";
+
+// Node17 and below treats header keys as case-sensitive, so the keys need to
+// be all lowercased for correct comparison.
```

Comment:
That's a pretty big issue for us, because we all methods of `Headers` to match names by case-insensitive byte sequence.

I'm actually a bit surprised because I would expect a lot of tests to fail if this is true.

Can you add tests for `Headers` to verify this? We can run tests on connect-core in both v16 and v18 (we should be doing this anyways), so we can have confidence.

Reply:
Yeah I'll add some.  I noticed this because the unit tests passed for me locally and then all failed in CI.  I was then able to recreate locally by switching to Node 16 and 17.   When I added this function in, they all seemed to pass in all Node versions.

I'm assuming that CI is using Node < 18 correct?